registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -463,35 +463,48 @@ public MessageDTO process(MessageDTO object) {
-		for (Map.Entry e : fieldMap.entrySet()) {
-			if (e.getValue() != null) {
-				String value = e.getValue().toString();
-				if (value != null) {
-					Object json = new JSONTokener(value).nextValue();
-					if (json instanceof org.json.JSONObject) {
-						HashMap<String, Object> hashMap = objectMapper.readValue(value, HashMap.class);
-						demographicIdentity.putIfAbsent(e.getKey(), hashMap);
-					}
-					else if (json instanceof JSONArray) {
-						List jsonList = new ArrayList<>();
-						JSONArray jsonArray = new JSONArray(value);
-						for (int i = 0; i < jsonArray.length(); i++) {
-							Object obj = jsonArray.get(i);
-							HashMap<String, Object> hashMap = objectMapper.readValue(obj.toString(), HashMap.class);
-							if(trimWhitespaces && hashMap.get("value") instanceof String) {
-								hashMap.put("value",((String)hashMap.get("value")).trim());
-							}
-							jsonList.add(hashMap);
-						}
-						demographicIdentity.putIfAbsent(e.getKey(), jsonList);
-					} else
-						demographicIdentity.putIfAbsent(e.getKey(), value);
-				} else
-					demographicIdentity.putIfAbsent(e.getKey(), value);
-			}
-		}
-	}
+    private void loadDemographicIdentity(Map<String, String> fieldMap, JSONObject demographicIdentity) throws IOException, JSONException {
+        for (Map.Entry e : fieldMap.entrySet()) {
+            if (e.getValue() == null) {
+                continue;
+            }
+
+            String value = e.getValue().toString();
+            if (value == null) {
+                demographicIdentity.putIfAbsent(e.getKey(), value);
+                continue;
+            }
+
+            Object json = new JSONTokener(value).nextValue();
+            if (json instanceof org.json.JSONObject) {
+                HashMap<String, Object> hashMap = objectMapper.readValue(value, HashMap.class);
+                demographicIdentity.putIfAbsent(e.getKey(), hashMap);
+                continue;
+            }
+
+            if (json instanceof JSONArray) {
+                List jsonList = new ArrayList<>();
+                JSONArray jsonArray = new JSONArray(value);
+                for (int i = 0; i < jsonArray.length(); i++) {
+                    Object obj = jsonArray.get(i);
+                    if (obj instanceof String) {
+                        jsonList.add(obj);
+                    } else {
+                        HashMap<String, Object> hashMap = objectMapper.readValue(obj.toString(), HashMap.class);
+
+                        if (trimWhitespaces && hashMap.containsKey("value") && hashMap.get("value") instanceof String) {
+                            hashMap.put("value", ((String) hashMap.get("value")).trim());
+                        }
+                        jsonList.add(hashMap);
+                    }
+                }
+                demographicIdentity.putIfAbsent(e.getKey(), jsonList);
+            }
+            else {
+                demographicIdentity.putIfAbsent(e.getKey(), value);
+            }
+        }

```

Comment:
Add the test case for respective changes.

Reply:
Added..